src/freenet/client/filter/MP3Filter.java

```diff
@@ -167,6 +170,29 @@ public void filter(InputStream input, OutputStream output) throws DataFilterExce
 					Logger.normal(this, "Lost sync for "+countLostSyncBytes+" bytes");
 				countLostSyncBytes = 0;
 				frameHeader = in.readInt();
+			} else if (!foundStream && (frameHeader & 0xffffff00) == 0x49443300) {
+				// This is an ID3v2 header, see http://id3.org/id3v2.3.0#ID3v2_header
+				// Skip minor version, flags
+				in.skip(2);
+				// ID3 tag size
+				byte[] encodedSize = new byte[4];
+				in.readFully(encodedSize);
+				int size = 0;
+				size |= (encodedSize[0] & 0x7F) << 21;
+				size |= (encodedSize[1] & 0x7F) << 14;
+				size |= (encodedSize[2] & 0x7F) << 7;
+				size |= (encodedSize[3] & 0x7F);
+				in.skip(size);
+				Logger.normal(this, "Skipped " + size + " bytes of ID3v2 data");
+				frameHeader = in.readInt();
+				foundStream = (frameHeader & 0xffe00000) == 0xffe00000;
+			} else if (!foundStream && (frameHeader & 0xffffff00) == 0x54414700) {
```

Comment:
ID3v1 headers are not located at the beginning of the file. Am I missing something here?